### PR TITLE
Add typings and format

### PR DESCRIPTION
**Pull request by Charliekenney23**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  isFinished(): Promise<Boolean>;
+}
+
+export interface CommitPathsFilter {
+  readonly added?: string[];
+  readonly modified?: string[];
+  readonly removed?: string[];
+}
+
+export interface BuildOptions {
```
Comment (mdlavin) [inline index.d.ts]: Is it really true that every field is optional? I would have expected something to be required

Comment (Charliekenney23) [reply index.d.ts]: Technically, the schema has no `required` properties. But the docs say:
> Specify an app slug and at least one parameter out of three: a git tag or git commit hash, a branch, or a workflow ID.

See: https://api-docs.bitrise.io/#/builds/build-trigger.

Perhaps I could add a type overload to extend from the base parameters and have at least one of those specified properties present. Or maybe a comment is sufficient?

**Comment by Charliekenney23** [comment]
> Are testcases needed?

I'll add an extra build test case spanning across all the params to assert they've been case transformed correctly.

**Comment by Charliekenney23** [comment]
@mdlavin I think I've addressed all of your feedback.

**Pull request merged**